src/main/scala/com/cognite/sdk/scala/common/ReadableResource.scala

```diff
@@ -129,11 +129,14 @@ object Readable {
 
 trait RetrieveByIds[R, F[_]] extends WithRequestSession[F] with BaseUrl {
   def retrieveByIds(ids: Seq[Long]): F[Seq[R]]
-  @SuppressWarnings(Array("org.wartremover.warts.IterableOps"))
   def retrieveById(id: Long): F[R] =
-    // The API returns an error causing an exception to be thrown if the item isn't found,
-    // so .head is safe here.
-    requestSession.map(retrieveByIds(Seq(id)), (r1: Seq[R]) => r1.head)
+    requestSession.map(
+      retrieveByIds(Seq(id)),
+      (r1: Seq[R]) =>
+        r1.headOption.getOrElse(
+          throw new SdkException("Unexpected empty response when retrieving item by Id")
```

Comment:
F.raiseError would probably look nicer, will see how simple is it to change

Reply:
Don't think it's that straightforward because r1.headOption is of type R, not F[R] since it's inside the requestSession.map which unwraps the F

It can probably still be done but not by simply adding implicit F MonadError and using F.raiseError

I think I would rather do that separately

Reply:
Yes, would need to do requestSession.flatMap, and add implicit ApplicativeError[F] argument (can't be a constraint on trait as far as I can see, so on trait method level)